Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -351,10 +351,20 @@ extension Processor {
     controller.pc = pc
     currentPosition = pos ?? currentPosition
     callStack.removeLast(callStack.count - stackEnd.rawValue)
-    storedCaptures = capEnds
     registers.ints = intRegisters
     registers.positions = posRegisters
 
+    if preservingCaptures {
+      for i in capEnds.indices {
+        if storedCaptures[i].range == nil {
+          storedCaptures[i].currentCaptureBegin = capEnds[i].currentCaptureBegin
```

Comment:
Do you have a test that fails if this line is commented out?

Reply:
You're right here – we can't have a capture that starts before a lookahead/atomic group but ends in the middle of it, which simplifies the `preservingCaptures` logic.